Sources/NIOEchoServer/main.swift

```diff
@@ -70,18 +70,25 @@ let defaultPort = 9999
 enum BindTo {
     case ip(host: String, port: Int)
     case unixDomainSocket(path: String)
+    case vsock(_: VsockAddress)
 }
 
 let bindTarget: BindTo
 switch (arg1, arg1.flatMap(Int.init), arg2.flatMap(Int.init)) {
-case (.some(let h), _ , .some(let p)):
-    /* we got two arguments, let's interpret that as host and port */
+case (_, .some(let cid), .some(let port)):
+    /* we got two arguments (Int, Int), let's interpret that as vsock cid and port */
+    bindTarget = .vsock(VsockAddress(
+        cid: .init(rawValue: UInt32(bitPattern: Int32(truncatingIfNeeded: cid))),
```

Comment:
nit, I'd suggest using the type name here instead of `.init` unless this is some cunning have the compiler pick the right, platform dependent type trick

Reply:
Done (Picked this habit up from another project which seemed to favour it).